### ISSUE TITLE
Extend derivative API (gradient, Hessian, multi-output partials)

### ISSUE THREAD

**Issue by vadmbertr** [f0uriest/interpax#143]
Hello again,

I have a use case (which I believe is quite common) where I need to evaluate multiple partial derivatives of a 2D or 3D field at the same point (e.g., f_x, f_y, f_xy, etc.).

With the current API, this requires multiple calls to `interp2d` or `interp3d` to obtain each partial derivative separately.

Would you consider extending the API to allow requesting multiple partial derivatives at once (possibly with mixed derivative orders)? This could enable returning the gradient and Hessian efficiently from a single interpolation operation.

Thank you again for your work!
Vadim

**Comment by f0uriest** [comment]
I think this should already be possible by vmapping over the `derivative` argument:
```python

x = np.linspace(0,1,10)
y = np.linspace(0,1,15)
z = np.linspace(0,1,20)
f = np.random.random((x.size, y.size, z.size))

xq = 0.5
yq = 0.5
zq = 0.5

ds = jnp.array([[1,0,0],[0,1,0],[0,0,1]])

grad = jax.vmap(lambda d: interpax.interp3d(xq,yq,zq,x,y,z,f,derivative=d, method="cubic"))(ds)

ds = jnp.array([[2,0,0],[1,1,0],[1,0,1],
                [1,1,0],[0,2,0],[0,1,1],
                [1,0,1],[0,1,1],[0,0,2],
               ])

hess = jax.vmap(lambda d: interpax.interp3d(xq,yq,zq,x,y,z,f,derivative=d, method="cubic"))(ds).reshape(3,3)

```
